jabgui/src/main/java/org/jabref/gui/walkthrough/WalkthroughAction.java

```diff
@@ -0,0 +1,113 @@
+package org.jabref.gui.walkthrough;
+
+import java.util.Map;
+import java.util.Objects;
+
+import javafx.scene.control.ContextMenu;
+
+import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.gui.frame.JabRefFrame;
+import org.jabref.gui.walkthrough.declarative.NavigationPredicate;
+import org.jabref.gui.walkthrough.declarative.NodeResolver;
+import org.jabref.gui.walkthrough.declarative.WindowResolver;
+import org.jabref.gui.walkthrough.declarative.effect.HighlightEffect;
+import org.jabref.gui.walkthrough.declarative.effect.MultiWindowHighlight;
+import org.jabref.gui.walkthrough.declarative.effect.WindowEffect;
+import org.jabref.gui.walkthrough.declarative.richtext.InfoBlock;
+import org.jabref.gui.walkthrough.declarative.richtext.TextBlock;
+import org.jabref.gui.walkthrough.declarative.step.PanelPosition;
+import org.jabref.gui.walkthrough.declarative.step.PanelStep;
+import org.jabref.gui.walkthrough.declarative.step.TooltipPosition;
+import org.jabref.gui.walkthrough.declarative.step.TooltipStep;
+import org.jabref.gui.walkthrough.declarative.step.WalkthroughStep;
+import org.jabref.logic.l10n.Localization;
+
+public class WalkthroughAction extends SimpleCommand {
+    private static final Map<String, Walkthrough> WALKTHROUGH_REGISTRY = buildRegistry();
```

Comment:
Calling complex methods in the static part of a class should not happen. Will increase loading times on startup and the pint in time this is called is not defined.
Refactor to lazy loading. Let's look into this in the call